ASM/c/dungeon_info.c

```diff
@@ -458,62 +479,126 @@ void draw_dungeon_info(z64_disp_buf_t *db) {
+                        left, top, icon_size, icon_size);
+            }
+        }
+
+        left += icon_size + padding;
+
+        if (CFG_DUNGEON_INFO_SILVER_RUPEES) {
+            // Draw silver rupee icons
+
+            sprite_load(db, &key_rupee_clock_sprite, 1, 1);
+
+            for (int i = 0; i < rows; i++) {
+                dungeon_entry_t *d = &(dungeons[d_right_dungeon_idx(i)]); // skip Deku/DC/Jabu/Ice dynamically
+                bool show_silver_rupees = false;
+                uint8_t *silver_rupee_puzzles = CFG_DUNGEON_IS_MQ[d->index] ? d->silver_rupee_puzzles_mq : d->silver_rupee_puzzles_vanilla;
+                for (int puzzle_idx = 0; puzzle_idx < 4; puzzle_idx++) {
+                    if (silver_rupee_puzzles[puzzle_idx] == (uint8_t) -1) break;
+                    uint8_t count = extended_savectx.silver_rupee_counts[silver_rupee_puzzles[puzzle_idx]];
+                    if (count > 0) {
+                        show_silver_rupees = true;
+                        break;
+                    }
+                }
+                if (show_silver_rupees) {
                     int top = start_top + ((icon_size + padding) * i);
-                    sprite_draw(db, &quest_items_sprite, 0,
+                    sprite_draw(db, &key_rupee_clock_sprite, 0,
                             left, top, icon_size, icon_size);
                 }
             }
 
-            left += icon_size + padding;
+            // Draw silver rupee counts
+            sprite_load(db, &font_sprite, 16, 10); // load characters 0 through 9
+
+            for (int i = 0; i < rows; i++) {
+                dungeon_entry_t *d = &(dungeons[d_right_dungeon_idx(i)]);
+                bool show_silver_rupees = false;
+                uint8_t *silver_rupee_puzzles = CFG_DUNGEON_IS_MQ[d->index] ? d->silver_rupee_puzzles_mq : d->silver_rupee_puzzles_vanilla;
+                for (int puzzle_idx = 0; puzzle_idx < 4; puzzle_idx++) {
+                    if (silver_rupee_puzzles[puzzle_idx] == (uint8_t) -1) break;
+                    uint8_t count = extended_savectx.silver_rupee_counts[silver_rupee_puzzles[puzzle_idx]];
+                    if (count > 0) {
+                        show_silver_rupees = true;
+                        break;
+                    }
+                }
+                if (show_silver_rupees) {
+                    int top = start_top + ((icon_size + padding) * i) + 1;
+                    for (int puzzle_idx = 0; puzzle_idx < 4; puzzle_idx++) {
+                        if (silver_rupee_puzzles[puzzle_idx] == (uint8_t) -1) break;
+                        silver_rupee_data_t var = silver_rupee_vars[silver_rupee_puzzles[puzzle_idx]][CFG_DUNGEON_IS_MQ[d->index]];
+                        uint8_t count = extended_savectx.silver_rupee_counts[silver_rupee_puzzles[puzzle_idx]];
+                        int puzzle_left = left + icon_size + padding + font_sprite.tile_w * (2 * puzzle_idx - 1);
```

Comment:
Additional padding between silver rupee counts is helpful for MQ Shadow where counts go up to 10. The suggestion is what I have on my branch. Also don't forget the A button menu with the different font size, ref. https://github.com/mracsys/OoT-Randomizer/blob/Dev-M/ASM/c/dungeon_info.c#L438
```suggestion
                        int puzzle_left = left + font_sprite.tile_w * (2 * puzzle_idx) + padding * 4 * puzzle_idx;
```